### PR TITLE
fix: circular dependencies in core

### PR DESCRIPTION
**Pull request by bathienle**
following #416, the refactoring from field injection to constructor injection introduces some circular dependencies.

I moved the problematic piece of code to a new service called ProjectMemberService which deals with adding or removing user from project

### CODE REVIEW

File: core/src/main/java/be/cytomine/controller/ontology/RestUserAnnotationController.java [hunk 1]
```diff
@@ -109,7 +109,9 @@ public void downloadDocumentByProject(
     ) throws IOException {
         Project project = projectService.find(idProject)
                 .orElseThrow(() -> new ObjectNotFoundException("Project", idProject));
-        users = userService.fillEmptyUserIds(users, idProject);
```
Comment (ldicarlo) [inline core/src/main/java/be/cytomine/controller/ontology/RestUserAnnotationController.java]: can you use a `Optional<String> users` here ?  So you don't need to test for null after
